frame/staking/src/offchain_election.rs

```diff
@@ -247,11 +458,141 @@ pub fn prepare_submission<T: Trait>(
 		}
 	}
 
-	// both conversions are safe; snapshots are not created if they exceed.
-	let size = ElectionSize {
-		validators: snapshot_validators.len() as ValidatorIndex,
-		nominators: snapshot_nominators.len() as NominatorIndex,
-	};
-
 	Ok((winners_indexed, compact, score, size))
 }
+
+#[cfg(test)]
+mod test {
+	#![allow(unused_variables)]
+	use super::*;
+	use crate::ElectionSize;
+
+	struct Staking;
+
+	impl crate::WeightInfo for Staking {
```

Comment:
This is going to be a pain to maintain. Can you not use the "default weights"?

Reply:
But then the test will be a function of (i.e. _break because of_) the _real_ staking weights, which might change. I think this is a pain both ways. At least now in the tests I can have my own simple implementation of WeightInfo which makes the test much more understandable than having to work with the real weight formula.